engine/runtime/src/main/java/org/enso/interpreter/node/expression/literal/IntegerLiteralNode.java

```diff
@@ -1,13 +1,18 @@
 package org.enso.interpreter.node.expression.literal;
 
+import com.oracle.truffle.api.Assumption;
+import com.oracle.truffle.api.CompilerDirectives;
+import com.oracle.truffle.api.Truffle;
 import com.oracle.truffle.api.frame.VirtualFrame;
 import com.oracle.truffle.api.nodes.NodeInfo;
 import org.enso.interpreter.node.ExpressionNode;
 
 /** A representation of integer literals in Enso. */
 @NodeInfo(shortName = "IntegerLiteral")
 public final class IntegerLiteralNode extends ExpressionNode {
-  private final long value;
+  private static final Assumption CONSTANTS_ARE_CONSTANTS = Truffle.getRuntime().createAssumption("Constants were never updated");
```

Comment:
A global assumption is too heavy handed. The changes are gonna keep happening in a single file, don't want to impact library code.

Reply:
Idea/plan: Let's introduce new node tag: `Patchable`. As soon as we have a _simple edit_ patch in a source file, we instrument all _patchable_ nodes in the source. Such instrumentation would de-opt, but only once. Each wrapper around _patchable_ would be able to hold an `Object value` and if non-null, return that one.

Subsequent full re-parses of once _patchable_ source file should produce AST ready to be _patched_ without de-opts.

Reply:
The _patchable idea_ has been implemented in [31648ad](https://github.com/enso-org/enso/pull/3508/commits/31648adb39548983b077f4bb0d0fe2ae3fd5fb9a). As soon as first patch is applied, the application de-opts, and instruments all `Patchable` nodes in the module source with an `ExecutionEventListener` that checks a map of _patched values_ and if a value is found, it returns it instead of running the original code. `IntegerLiteralNode` has been modified to implement `Patchable`.